frontend/app/embed.ts

```diff
@@ -47,85 +48,98 @@ function createFrame({
+  window.REMARK42 = window.REMARK42 || {};
+  window.REMARK42.createInstance = createInstance;
 
-  if (!node) {
-    console.error("Remark42: Can't find root node.");
-    return;
-  }
+  if (window.remark_config) {
+    const remark42Instance = createInstance(window.remark_config);
 
-  try {
-    window.remark_config = window.remark_config || {};
-  } catch (e) {
-    console.error('Remark42: Config object is undefined.');
-    return;
+    if (remark42Instance) {
+      window.REMARK42.changeTheme = remark42Instance.changeTheme;
+      window.REMARK42.destroy = remark42Instance.destroy;
+    }
   }
 
-  if (!window.remark_config.site_id) {
-    console.error('Remark42: Site ID is undefined.');
-    return;
-  }
+  function createInstance(remark_config: CommentsConfig) {
+    let initDataAnimationTimeout: NodeJS.Timeout | null = null;
 
-  window.remark_config.url = (window.remark_config.url || window.location.origin + window.location.pathname).split(
-    '#'
-  )[0];
+    try {
+      remark_config = remark_config || {};
+    } catch (e) {
+      console.error('Remark42: Config object is undefined.');
+      return;
+    }
 
-  window.REMARK42 = window.REMARK42 || {};
-  window.REMARK42.changeTheme = changeTheme;
+    if (!remark_config.site_id) {
+      console.error('Remark42: Site ID is undefined.');
+      return;
+    }
 
-  const query = Object.keys(window.remark_config)
-    .filter(key => key !== '__colors__')
-    .map(key => {
-      return `${encodeURIComponent(key)}=${encodeURIComponent(
-        window.remark_config[key as keyof typeof window.remark_config]
-      )}`;
-    })
-    .join('&');
-  const iframe = createFrame({ host: BASE_URL, query, __colors__: window.remark_config.__colors__ });
+    remark_config.url = (remark_config.url || window.location.origin + window.location.pathname).split('#')[0];
 
-  node.appendChild(iframe);
+    const node =
+      remark_config.node instanceof HTMLElement
+        ? remark_config.node
+        : document.getElementById(remark_config.node || NODE_ID);
 
-  window.addEventListener('message', receiveMessages);
-  window.addEventListener('hashchange', postHashToIframe);
-  document.addEventListener('click', postClickOutsideToIframe);
+    if (!node) {
+      console.error("Remark42: Can't find root node.");
+      return;
+    }
 
-  const titleElement = document.querySelector('title');
-  if (titleElement) {
-    new MutationObserver(mutations => postTitleToIframe(mutations[0].target.textContent!)).observe(titleElement, {
-      subtree: true,
-      characterData: true,
-      childList: true,
-    });
-  }
+    const query = Object.keys(remark_config)
+      .filter(key => key !== '__colors__')
+      .map(key => {
+        return `${encodeURIComponent(key)}=${encodeURIComponent(remark_config[key as keyof typeof remark_config])}`;
+      })
+      .join('&');
+    const iframe =
+      (document.querySelector('iframe[title=Remark42]') as HTMLIFrameElement) ||
```

Comment:
You don't need to check that frame already exist. You could do it in your integration code and don't create new instance with the same config.